### PR TITLE
Inline Diagnostics - add null check

### PR DESCRIPTION
**Pull request by akhera99**
Adds a null check in case the TextViewLineCollection does not contain the specified buffer position.

### CODE REVIEW

File: src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs [hunk 1]
```diff
@@ -152,6 +152,11 @@ protected override void AddAdornmentsToAdornmentLayer_CallOnlyOnUIThread(Normali
 
                     var viewLine = viewLines.GetTextViewLineContainingBufferPosition(mappedPoint);
 
+                    if (viewLine is null)
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs:155]: needs docs on when this could happen.

Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs:155]: could we push this logic into ShouldDrawTag?

Comment (Youssef1313) [reply src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs:155]: Is the scenario where this is reachable applicable to this call?

https://github.com/dotnet/roslyn/blob/605765f4c36a24533f2f5340ad569e1d81c92763/src/EditorFeatures/Core.Wpf/StringIndentation/StringIndentationAdornmentManager.VisibleBlock.cs#L55-L56

**Comment by Youssef1313** [comment]
> in case the TextViewLineCollection does not contain the specified buffer position.

https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.text.editor.iwpftextview.gettextviewlinecontainingbufferposition?view=visualstudiosdk-2022

The documentation says that it throws `ArgumentOutOfRangeException` if `bufferPosition` is not a valid buffer position and mentions nothing about returning null.

Am I misinterpreting it? or is the documentation outdated?

**Comment by akhera99** [comment]
> > in case the TextViewLineCollection does not contain the specified buffer position.
> 
> https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.text.editor.iwpftextview.gettextviewlinecontainingbufferposition?view=visualstudiosdk-2022
> 
> The documentation says that it throws `ArgumentOutOfRangeException` if `bufferPosition` is not a valid buffer position and mentions nothing about returning null.
> 
> Am I misinterpreting it? or is the documentation outdated?

I believe the documentation is just outdated.

File: src/EditorFeatures/Core.Wpf/Adornments/AbstractAdornmentManager.cs [hunk 2]
```diff
@@ -251,16 +252,26 @@ protected void UpdateSpans_CallOnlyOnUIThread(NormalizedSnapshotSpanCollection c
...
 
+            var mappedPoint = point.Value;
+            viewLine = TextView.TextViewLines.GetTextViewLineContainingBufferPosition(mappedPoint);
+
+            // Unsure what the scenario is - but sometimes the SnapshotPoint is not located
+            // within any of the lines in the TextViewLineCollection. In that case, we do not want to draw a tag.
+            if (viewLine is null)
+            {
+                return false;
+            }
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/Adornments/AbstractAdornmentManager.cs:273]: did you validate that this line-separates and raw-string-indentation-tags still show up as expected?

Comment (akhera99) [reply src/EditorFeatures/Core.Wpf/Adornments/AbstractAdornmentManager.cs:273]: Yep, they seem to show up the same as they did before

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/EditorFeatures/Core.Wpf/Adornments/AbstractAdornmentManager.cs [hunk 3]
```diff
@@ -251,16 +252,26 @@ protected void UpdateSpans_CallOnlyOnUIThread(NormalizedSnapshotSpanCollection c
...
             if (point is null)
             {
                 return false;
             }
 
+            var mappedPoint = point.Value;
+            viewLine = TextView.TextViewLines.GetTextViewLineContainingBufferPosition(mappedPoint);
+
+            // Unsure what the scenario is - but sometimes the SnapshotPoint is not located
+            // within any of the lines in the TextViewLineCollection. In that case, we do not want to draw a tag.
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core.Wpf/Adornments/AbstractAdornmentManager.cs:269]: @akhera99 @CyrusNajmabadi Did we reach out to the editor team on this one? I think is this a case where the line is in the middle of layout and they haven't completed it yet?

**Pull request merged**